draft-ietf-tls-tls13.md

```diff
@@ -2708,9 +2710,14 @@ suite, if available.
 If the server selects a PSK cipher suite, it MUST send a
 "pre_shared_key" extension with the identity that it selected.
 The client MUST verify that the server's selected_identity
-is within the range supplied by the client. If any other value
-is returned, the client MUST generate a fatal
-"unknown_psk_identity" alert and close the connection.
+is within the range supplied by the client. If the server supplies an
+"early_data" extension, the client MUST verify that the server selected the
+first offered identity. If any other value is returned, the client MUST
+generate a fatal "unknown_psk_identity" alert and close the connection.
```

Comment:
This actually suggests an alternative possibility, which is that the selected PSK is used in place of the (EC)DHE secret for pure PSK suites and that the selection doesn't matter when PSK is only used for 0-RTT.  I guess that your position is that this is complicated and we don't need that complication.  I agree.